### ISSUE TITLE
Changes between groovy-eclipse-batch 2.4.3-01 vs 2.4.16-02

### ISSUE THREAD

**Issue by kaushalkumar** [groovy/groovy-eclipse#1157]
My project builds successfully with groovy-eclipse-compiler, but fails without groovy-eclipse-compiler (using just javac). The build fails with an error message as given below (reported in a test class, while mocking an invocation)
`java: reference to getFileResource is ambiguous`
In order to debug the issue, I created a project with minimal files which is available at [project-debug](https://github.com/kaushalkumar/project-debug). Though in project we have groovy source also, but I have not included them here to keep the code minimal. 

The reported issue looks to be legitimate and I feel that groovy-eclipse-compiler must also fail, but it seems that the error is ignored. We are using _Java version - 1.8.0_231, Maven - 3.6.2, maven-compiler-plugin -3.3, groovy-eclipse-compiler - 2.9.2-01, groovy-eclipse-batch - 2.4.3-01._

I posted this query on [question on stackoverflow](https://stackoverflow.com/questions/63846102/groovy-eclipse-compiler-compiles-but-javac-compilation-fails) and got a suggestion to upgrade and the jar versions. Upgrading groovy-eclipse-batch jar to 2.4.16-02 helps and reports the compilation issues, whereas 2.4.3-01 does not.

I am trying to figure out the ticket against which this issue was fixed. I scanned the repository for couple of hours, but was not able to find it, so am raising a request here.

Though this is not a high priority, but some insight in this area will help me to understand the issue and its fix.

Thanks for your help in advance.

Regards,
Kaushal

**Comment by eric-milles** [comment]
This history of this file will let you know when each version of the batch compiler artifact was built:  https://github.com/groovy/groovy-eclipse/blob/master/extras/groovy-eclipse-batch-builder/build.properties

You can also look at the history of the `jdt.patch.target` property in that file to see which JDT (Java Development Tools) version was used as the base.  Eclipse keeps history for the JDT at https://github.com/eclipse/eclipse.jdt.core, in case you are looking for a specific change that altered this behavior in the compiler.

**Comment by kaushalkumar** [comment]
Hi @eric-milles  - Thanks for providing input. Based on your comment, I explored the history of build.properties and found that the compilation issue was between 2.4.12-01 (works) and 2.4.12-02 (breaks- as expected), which was part of release 2.9.2.

The change happened on Aug 10, 2017 ([https://github.com/groovy/groovy-eclipse/commit/13c1c2ae839380b948808dc7028486fd596fa753#diff-c8c111c3afb6080ae6b32148caaf6a0a](https://github.com/groovy/groovy-eclipse/commit/13c1c2ae839380b948808dc7028486fd596fa753#diff-c8c111c3afb6080ae6b32148caaf6a0a)), with comment as "Remove codehaus references". The _jdt.patch.target_ was targeted for e44 which is Luna. This was same for both the files.

I invested some time in exploring https://github.com/eclipse/eclipse.jdt.core, to figure out the how compiler behaviour could have altered, but could not get much. Though I am not very sure, but I feel that change in groovy-eclipse-batch (between 2.4.12-01 and 2.4.12-02) might be the cause of this.

Having invested this much time , I feel that it is not worth to further debug on this to get the root cause as it is already fixed in next version(s).

Thanks for your help. I appreciate it.

Cheers
